contracts/collections/sg721-base/src/contract.rs

```diff
@@ -229,27 +229,57 @@ where
             .royalty_info
             .unwrap_or_else(|| current_royalty_info.clone());
 
-        // TODO: get the factory from the minter
-        // TODO: convert bps to percentage if needed
+        // get the factory from the minter to get
+        // max_royalty, max_royalty_increase_rate, royalty_min_time_duration_secs
+        let minter_address = self.parent.minter(deps.as_ref())?.minter.unwrap();
+        let minter_config: MinterConfig<T> = deps
+            .querier
+            .query_wasm_smart(minter_address, &MinterQueryMsg::Config {})?;
+        let factory_address = minter_config.factory;
+        let factory_params: ParamsResponse<T> = deps
+            .querier
+            .query_wasm_smart(factory_address, &Sg2QueryMsg::Params {})?;
+        let max_royalty = factory_params.params.max_royalty_bps.bps_to_decimal();
+        let max_royalty_increase_rate = factory_params
+            .params
+            .max_royalty_increase_rate_bps
+            .bps_to_decimal();
+        let royalty_min_time_duration = factory_params.params.royalty_min_time_duration_secs;
 
         // reminder: collection_msg.royalty_info is Option<Option<RoyaltyInfoResponse>>
-        collection.royalty_info = if let Some(royalty_info) = new_royalty_info {
+        collection.royalty_info = if let Some(new_royalty_info_res) = new_royalty_info {
+            // check if new_royalty_info_res > max_royalty
+            if new_royalty_info_res.share > max_royalty {
+                return Err(ContractError::RoyaltyShareTooHigh {});
+            }
             // update royalty info to equal or less, else throw error
-            if let Some(royalty_info_res) = current_royalty_info {
-                // TODO: if royalty_info.share > royalty_info_res.share + factory.royalty_share_increase_rate
-                // TODO: also check against the `royalties_updated_at` timestamp with the current block time
-                if royalty_info.share > royalty_info_res.share {
+            if let Some(curr_royalty_info_res) = current_royalty_info {
+                if new_royalty_info_res.share
+                    > curr_royalty_info_res.share + max_royalty_increase_rate
+                {
                     return Err(ContractError::RoyaltyShareIncreasedTooMuch {});
                 }
+
+                // if royalty share changed and current time is after last royalty update + min duration
+                // royalty_updated_at is always Some because it is set in instantiate
+                if (new_royalty_info_res.share != curr_royalty_info_res.share)
+                    && (curr_royalty_info_res
+                        .updated_at
+                        .plus_seconds(royalty_min_time_duration)
+                        > env.block.time)
```

Comment:
Should this be `<`?

Reply:
`if curr royalty updated_at + 24 hours > block time, throw error` I think that's right.